### PR TITLE
Add isAdmin to getCurrentUser

### PR DESCRIPTION
**Pull request by georgehrke**
Requires https://github.com/nextcloud/nextcloud-typings/pull/34/files

### CODE REVIEW

**Comment by ChristophWurst** [comment]
> Requires https://github.com/nextcloud/nextcloud-typings/pull/34/files

Done. I'll push a release once https://github.com/nextcloud/nextcloud-typings/pull/35 has been integrated.

File: lib/user.ts [hunk 1]
```diff
@@ -1,3 +1,7 @@
+/// <reference types="@nextcloud/typings" />
+
+declare var OC: Nextcloud.v16.OC | Nextcloud.v17.OC | Nextcloud.v18.OC;
```
Comment (ChristophWurst) [inline lib/user.ts:3]: - [ ] TODO: use v19 as well https://github.com/nextcloud/nextcloud-typings/pull/35

File: lib/user.ts [hunk 2]
```diff
@@ -6,9 +10,17 @@ const displayNameElement = document
 	.getElementsByTagName('head')[0]
 const displayName = displayNameElement ? displayNameElement.getAttribute('data-user-displayname') : null
 
+let isAdmin
```
Comment (ChristophWurst) [inline lib/user.ts:13]: nit: use ternary operator to avoid reassignment and use `const` instead

File: lib/user.ts [hunk 3]
```diff
@@ -6,9 +10,14 @@ const displayNameElement = document
 	.getElementsByTagName('head')[0]
 const displayName = displayNameElement ? displayNameElement.getAttribute('data-user-displayname') : null
 
+const isAdmin = (typeof OC === 'undefined')
+	? false
+	: OC.isUserAdmin()
```
Comment (ChristophWurst) [inline lib/user.ts:15]:
```
lib/user.ts(15,7): error TS2339: Property 'isUserAdmin' does not exist on type 'Nextcloud.v16.OC | Nextcloud.v17.OC | Nextcloud.v18.OC | Nextcloud.v19.OC'.
```

Comment (ChristophWurst) [reply lib/user.ts:15]: My fault. I did not understand why you added it to the common and v17 typings. But this is actually a bit of duplicated code due to minor differences … https://github.com/nextcloud/nextcloud-typings/pull/34#discussion_r433953512 was a mistake. Could you send a PR to fix that?

Sorry

Comment (georgehrke) [reply lib/user.ts:15]: Sure, will do. No worries

File: lib/user.ts [hunk 4]
```diff
@@ -6,9 +10,14 @@ const displayNameElement = document
...
+
 export interface NextcloudUser {
```
Comment (georgehrke) [inline lib/user.ts:17]: Should we also just extend this from `Nextcloud.Common.CurrentUser`?

File: lib/user.ts (hunk 3 above)
Comment (ChristophWurst) [reply lib/user.ts:15]: v0.2.2 of the typings should work :pray:

File: lib/user.ts (hunk 4 above)
Comment (ChristophWurst) [reply lib/user.ts:17]: Sure!

Comment (georgehrke) [reply lib/user.ts:17]: Actually nevermind. `Nextcloud.Common.CurrentUser` allows `uid` to be false, which will never be the case here.

File: lib/user.ts (hunk 3 above)
Comment (georgehrke) [reply lib/user.ts:15]: Updated

File: lib/user.ts (hunk 4 above)
Comment (ChristophWurst) [reply lib/user.ts:17]: Right, there was something :stuck_out_tongue_closed_eyes:

**Comment by ChristophWurst** [comment]
v1.3 released :shipit:

**Pull request merged**